### PR TITLE
Update changelog

### PR DESCRIPTION
**Pull request by lencioni**
Some of the commit messages in the git log were a bit hard to follow, so
I'm not sure I got it all in correctly, but it is definitely better than
what was here before.

### CODE REVIEW

**Comment by trotzig** [comment]
I think you got it all. I actually forgot that we kept this log.

**Comment by lencioni** [comment]
Thanks for merging! This is really helpful for people who want to upgrade without breaking things.

**Comment by trotzig** [comment]
Cool. Let me know if you run into any issues after upgrading!
